File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-27.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) {
        let count = {};
        let l = 0;
        let res = 0;
        let maxF = 0;
        for (let r = 0; r < s.length; r ++) {
            let cur = s[r];
            count[cur] = ( count[cur] || 0 ) + 1;
            maxF = Math.max(count[cur], maxF)
            let curLen = r - l + 1;
            while(curLen - maxF > k) {
                let deleted = s[l];
                count[deleted] --;
                l ++;
                curLen = r - l + 1;
            }

            curLen = r - l + 1;
            res = Math.max(curLen, res)
        }

        return res
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-27.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        cur = 0
        curMax =  0 
        jump = 0 
        

        for i in range(len(nums)): 
            if i > cur:
                cur = curMax 
                jump += 1 

            curMax = max(curMax, i + nums[i]) 

        return jump